src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj

```diff
@@ -139,25 +139,29 @@
   </Target>
 
   <Target Name="CreatePackNupkg">
+    <PropertyGroup>
+      <!-- Build from source can't use ILMerge. -->
+      <ILMergeSubpath Condition="'$(DotNetBuildFromSource)' != 'true'">ilmerge\</ILMergeSubpath>
```

Comment:
Use `IsBuildOnlyXPLATProjects`. 

It defaults to the `DotNetBuildFromSource` value.

Reply:
Ok. I commented on this in the description with my reasoning (copying below just in case you didn't see it, not sure from your comment). I'm fine either way so I'll change it.

> Only use ILMerge paths when `DotNetBuildFromSource != true`. It would also be fine to use `IsBuildOnlyXPLATProjects`. `DotNetBuildFromSource` seems more precise (in theory someone could build only xplat projects but still use ILMerge?), but let me know.